### ISSUE TITLE
Parameters starting with "-"

### ISSUE THREAD

**Issue by mathiasrw** [yargs/yargs#832]
Thanks for an awesome project. 

I am seeking to invoke a program using yargs with two strings as parameters. Working example: 

```js
let argv = require('yargs')
	.strict()
	.help('h')
	.argv	
;

console.log(argv._[0],argv._[1]);
```

Lets call the program foobar. If I call foobar with a string that starts with `-` the rest of the string will be considered parameters

```bash
> foobar "header" "- list"
Options:
  -h           Show help                                                [boolean]

Unknown arguments:  , l, i, s, t
```

In the beginning I suspected it was a bug, but as this could be replicated in version 7,6,5,4,3 of yargs I investigated a bit more and found that there is no way to tell from `process.argv;` if an argument was wrapped in `"..."` or not - so way to seperate flags from string parameters (or: they are all "just" string parameters)

I could remove the `strict()` part, but then the program wont be able to access argument 2 (as it has been parsed as l,i,s,t flags).

Is there any way to tell yargs that argument number `n` and `m` is to be a string and not flags?

If not, I imagine I will need to force my users to always put flags after the second parameter, and use the yargs parser after I have removed the two first parameters. 

Any inputs very welcome!

**Comment by bcoe** [comment]
@mathiasrw I think part of the problem is that by the time yargs gets `argv`, it will have stripped the quotes so you will end up with:

```
foobar header - list
```

It feels a bit weird that this gets treated as `-list`, part of me would expect this to be parsed as:

`{_: ["header', "-", "list"]}`

What do you think the ideal behavior would be, given that it's probably impossible for yargs to know that it was passed "- list".

**Comment by bcoe** [comment]
@mathiasrw did this help clarify things? may I close this ticket.

**Comment by mathiasrw** [comment]
Ohh, man, sorry. Missed the notification about your first answer!

Intuitively `{_: ["header', "-", "list"]}` would be the "right" solution, but the space between `-` and the option might be a tradition in the Unix world. 

Maybe it could be a configuration for yargs if multi-single-option is supported or if a single '-' would indicate that the following is an option (demand there is no space after the `-`)

**For others who find themselves in same dilemma**: 
If you have mandatory parameters that can start with ´-` there are a few ways to deal with it. 

- Ask the user to place all options before a `--` and have the rest of the parameters after the `--` (everything after will not be considered options) 

- Have all parameters given by an option and ask the user to use the notation `'-q=- list'` 

- Start manipulating `process.argv` before yargs get to it.  

The last solution seems tempting but can be tricky. I ended up on this one while making a CLI search-and-replace tool. It needs two parameters (what to look for and what to replace it with). The core of my solution is to state that no options can be given before the two first parameters and then to do a `let [pattern, replacement] = process.argv.splice(2,2)` (ES6 notation) before calling yargs, so that yargs does not handle the parameters at all. 

But then, off cause, your need to handle cases where the user provides too few parameters. And after implementing the naive "I just tell the user to use minimum two parameters and exit" you realise you would like to do responding with the yargs generated help text - only available after you have done yargs. So I ended up doing a bit of fiddling both [before yargs](https://github.com/mathiasrw/rexreplace/blob/5632c5aaa6547d905a1bbdbedcda5d202a145cae/src/rexreplace.cli.js#L10-L15) and [after yargs](https://github.com/mathiasrw/rexreplace/blob/5632c5aaa6547d905a1bbdbedcda5d202a145cae/src/rexreplace.cli.js#L140-L142). 

You think you nailed it now, but after a few days in production, you realise you broke toe ability to only give the `-v` parameter - and realise you must let yargs run on all of the parameters in all cases where your criteria of minimum amount of parameters are not met. 

**Long story short**
I ended up with [this code before yargs](https://github.com/mathiasrw/rexreplace/blob/f09fb4d5ba35f5b678b7c909e1f000527981e799/src/cli.js#L7-L13) letting yargs deal with cases like `-v` and `-help` (where the program should not exit with error) and having [this code after yargs](https://github.com/mathiasrw/rexreplace/blob/f09fb4d5ba35f5b678b7c909e1f000527981e799/src/cli.js#L150-L154) making sure the program displays the help and exits with an error.
